std/string.d

```diff
@@ -2638,6 +2638,18 @@ auto lineSplitter(KeepTerminator keepTerm = KeepTerminator.no, Range)(Range r)
     return LineSplitter!(keepTerm, Range)(r);
 }
 
+///
+@safe pure unittest
+{
+    string s = "Hello\nmy\rname\nis";
+    string[] split = splitLines(s);
+
+    int idx = 0;
+    foreach(it; lineSplitter(s)) {
```

Comment:
Why not `foreach(idx, it; s.lineSplitter.enumerate)` and get rid of the separate `idx`?